enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_any?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_none?
+    i = 0
+    while i < size
+      return false if yield(self[i])
+
+      i += 1
+    end
+    true
+  end
+
+  def my_count
+    i = 0
+    count = []
+    if block_given?
+      while i < size
+        count << self[i] if yield(self[i])
+        i += 1
+      end
+      return count.size
+    end
+    size
+  end
+
+  def my_map(&block)
+    arr = []
+    my_each do |e|
+      arr << block.call(e)
+    end
+    arr
+  end
+
+  def my_inject(initial = nil)
+    result = initial.nil? ? self[0] : initial
+    (1..length - 1).each do |i|
+      result = yield(result, self[i])
+    end
+    result
+  end
+end
```

Comment:
- You should use 'my_each' here instead of the original each method

- my_inject combines all elements of enum by applying a binary operation, specified by a block
`[1,2,3,4].inject { |x,y| x + y }`
`=> 10`

- When a symbol is specified combines each element of the collection by applying the symbol as a named method:
`[1,2,3,4].inject (:+)`
`=> 10`
`[1,2,3,4].inject (:-)`
`=> -8`
`[1,2,3,4].inject (:*)`
`=> 24`